lms/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -0,0 +1,164 @@
+from .. import views
+
+
+class _DispatchingViewTestCase(TestCase):
+    """
+    Base class for tests that exercise DispatchingViews.
+    """
+    dop_adapter = adapters.DOPAdapter()
+    dot_adapter = adapters.DOTAdapter()
+
+    view_class = None
+    url = None
+
+    def setUp(self):
+        super(_DispatchingViewTestCase, self).setUp()
+        self.user = UserFactory()
+        self.dot_app = self.dot_adapter.create_public_client(
+            name='test dot application',
+            user=self.user,
+            redirect_uri='https://example.edx/redirect',
+            client_id='dot-app-client-id',
+        )
+        self.dop_client = self.dop_adapter.create_public_client(
+            name='test dop client',
+            user=self.user,
+            redirect_uri='https://example.edx/redirect',
+            client_id='dop-app-client-id',
+        )
+
+    def _post_request(self, user, client):
+        """
+        Call the view with a POST request objectwith the appropriate format,
+        returning the response object.
+        """
+        return self.client.post(self.url, self._post_body(user, client))
+
+    def _post_body(self, user, client):
+        """
+        Return a dictionary to be used as the body of the POST request
+        """
+        raise NotImplementedError()
+
+
+@ddt.ddt
+class TestAccessTokenView(_DispatchingViewTestCase):
+    """
+    Test class for AccessTokenView
+    """
+
+    view_class = views.AccessTokenView
+    url = reverse('access_token')
+
+    def _post_body(self, user, client):
+        """
+        Return a dictionary to be used as the body of the POST request
+        """
+        return {
+            'client_id': client.client_id,
+            'grant_type': 'password',
+            'username': user.username,
+            'password': 'test',
+        }
+
+    @ddt.data('dop_client', 'dot_app')
+    def test_access_token_fields(self, client_attr):
+        client = getattr(self, client_attr)
+        response = self._post_request(self.user, client)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertIn('access_token', data)
+        self.assertIn('expires_in', data)
+        self.assertIn('scope', data)
+        self.assertIn('token_type', data)
+
+    def test_dot_access_token_provides_refresh_token(self):
+        response = self._post_request(self.user, self.dot_app)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertIn('refresh_token', data)
+
+    def test_dop_public_client_access_token(self):
+        response = self._post_request(self.user, self.dop_client)
+        self.assertEqual(response.status_code, 200)
+        data = json.loads(response.content)
+        self.assertNotIn('refresh_token', data)
+
+
+@ddt.ddt
+@httpretty.activate
+class TestAccessTokenExchangeView(ThirdPartyOAuthTestMixinGoogle, ThirdPartyOAuthTestMixin, _DispatchingViewTestCase):
```

Comment:
I don't think it needs to be.  The purpose of these tests is to ensure that the dispatch view dispatches correctly to the actual view.  The behavior of that view is tested more thoroughly in the auth-exchange app.  Adding more tests here will just duplicate the work done there.